cpu/samd5x/cpu.c

```diff
@@ -203,4 +203,9 @@ void cpu_init(void)
 
     /* trigger static peripheral initialization */
     periph_init();
+
+    /* set ONDEMAND bit after all clocks have been configured */
+    /* This is to avoid setting the source for the main clock to ONDEMAND before using it. */
+    OSCCTRL->DFLLCTRLA.reg |= OSCCTRL_DFLLCTRLA_ONDEMAND;
+    OSCCTRL->Dpll[0].DPLLCTRLA.reg |= OSCCTRL_DPLLCTRLA_ONDEMAND;
```

Comment:
I am wondering if setting this here is mandatory.
I think this should work the same if you set this bit and ENABLE at the same time.

Reply:
I though so too, but when I set the `ONDEMAND` bit before connecting the clock to something, the CPU will stop when I then use that clock for the main clock.

Reply:
Huh, good point.

Reply:
But I agree that it is somewhat ugly.